src/app/item-triage/ItemTriage.tsx

```diff
@@ -0,0 +1,411 @@
+            <React.Fragment key={stat.statHash}>
+              <div className={styles.bestStat}>
+                <span className={styles.statIconWrapper}>
+                  {(stat.displayProperties.icon && (
+                    <BungieImage
+                      key={stat.statHash}
+                      className={clsx(styles.inlineIcon, styles.smaller)}
+                      src={stat.displayProperties.icon}
+                    />
+                  )) ||
+                    ' '}
+                </span>
+                <span className={styles.statValue}>{best}</span>{' '}
+                <span className={styles.dimmed}>{stat.displayProperties.name}</span>
+              </div>
+              <div className={styles.thisStat}>
+                <span className={styles.statValue}>{stat.base}</span> (
+                <span style={{ color: getValueColors(quality)[1] }}>{percent}%</span>)
+              </div>
+              <div className={styles.keepMeter}>
+                <KeepJunkDial value={quality} />
+              </div>
+            </React.Fragment>
+          ))}
+          {item.bucket.inArmor && (
+            <>
+              <div className={styles.bestStat}>
+                <span className={styles.statIconWrapper}> </span>
+                <span className={styles.statValue}>{notableStats.customTotalMax.best}</span>{' '}
+                <StatTotalToggle forClass={item.classType} className={styles.inlineBlock} />
+              </div>
+              <div className={styles.thisStat}>
+                <span className={styles.statValue}>{notableStats.customTotalMax.stat}</span> (
+                <span style={{ color: getValueColors(notableStats.customTotalMax.quality)[1] }}>
+                  {notableStats.customTotalMax.percent}%
+                </span>
+                )
+              </div>
+              <div className={styles.keepMeter}>
+                <KeepJunkDial value={notableStats.customTotalMax.quality} />
+              </div>
+            </>
+          )}
+        </div>
+      )}
+    </div>
+  );
+}
+
+/**
+ * for all items relevant for comparison to the seed item, processes them into a Record,
+ * keyed by item factor combination i.e. "arcwarlockopulent"
+ * with values representing how many of that type you own
+ */
+function collectRelevantItemFactors(exampleItem: D2Item) {
+  const combinationCounts: { [key: string]: number } = {};
+  getAllItems(exampleItem.getStoresService().getStores())
+    .filter(
+      (i) =>
+        // compare only items with the same canonical bucket.
+        i.bucket.sort === exampleItem.bucket.sort &&
+        // accept anything if seed item is class unknown
+        (exampleItem.classType === DestinyClass.Unknown ||
+          // or accept individual items if they're matching or unknown.
+          i.classType === DestinyClass.Unknown ||
+          i.classType === exampleItem.classType)
+    )
+    .forEach((item: D2Item) => {
```

Comment:
You *might* want to try unwrapping this and the filter into a for-loop and see if that helps performance noticeably.

Reply:
will investigate